sections/dashboard/Stake/InputCards/EscrowInputCard.tsx

```diff
@@ -1,112 +1,115 @@
-			!isNil(escrowedBalance) && escrowedBalance.gt(0)
-				? escrowedBalance.sub(stakedEscrowedBalance ?? zeroBN) ?? zeroBN
-				: zeroBN,
-		[escrowedBalance, stakedEscrowedBalance]
+			escrowedKwentaBalance.gt(0) ? escrowedKwentaBalance.sub(stakedEscrowedKwentaBalance) : zeroBN,
+		[escrowedKwentaBalance, stakedEscrowedKwentaBalance]
 	);
 
-	const handleTabChange = useCallback((tabIndex: number) => {
-		setAmount('');
-		setActiveTab(tabIndex);
-	}, []);
-
 	const maxBalance = useMemo(
-		() =>
-			activeTab === 0
-				? wei(unstakedEscrowedKwentaBalance ?? zeroBN)
-				: wei(stakedEscrowedBalance ?? zeroBN),
-		[activeTab, stakedEscrowedBalance, unstakedEscrowedKwentaBalance]
+		() => (activeTab === 0 ? wei(unstakedEscrowedKwentaBalance) : wei(stakedEscrowedKwentaBalance)),
+		[activeTab, stakedEscrowedKwentaBalance, unstakedEscrowedKwentaBalance]
 	);
 
-	const onMaxClick = useCallback(async () => {
+	const stakeEnabled = useMemo(() => {
+		return activeTab === 0 && unstakedEscrowedKwentaBalance.gt(0) && !!parseFloat(amount);
+	}, [activeTab, unstakedEscrowedKwentaBalance, amount]);
+
+	const unstakeEnabled = useMemo(() => {
+		return activeTab === 1 && stakedEscrowedKwentaBalance.gt(0) && !!parseFloat(amount);
+	}, [activeTab, stakedEscrowedKwentaBalance, amount]);
+
+	const handleApprove = useCallback(() => {
+		dispatch(approveKwentaToken('kwenta'));
+	}, [dispatch]);
+
+	const handleStakeEscrow = useCallback(() => {
+		dispatch(stakeEscrow(amountBN));
+	}, [dispatch, amountBN]);
+
+	const handleUnstakeEscrow = useCallback(() => {
+		dispatch(unstakeEscrow(amountBN));
+	}, [dispatch, amountBN]);
+
+	const onMaxClick = useCallback(() => {
 		setAmount(truncateNumbers(maxBalance, DEFAULT_TOKEN_DECIMALS));
 	}, [maxBalance]);
 
-	const { config: stakedEscrowKwentaConfig } = usePrepareContractWrite({
-		...rewardEscrowContract,
-		functionName: 'stakeEscrow',
-		args: [amountBN],
-		overrides: {
-			gasLimit: STAKING_LOW_GAS_LIMIT,
-		},
-		enabled: activeTab === 0 && unstakedEscrowedKwentaBalance.gt(0) && !!parseFloat(amount),
-	});
-
-	const { config: unstakedEscrowKwentaConfig } = usePrepareContractWrite({
-		...rewardEscrowContract,
-		functionName: 'unstakeEscrow',
-		args: [amountBN],
-		overrides: {
-			gasLimit: STAKING_LOW_GAS_LIMIT,
-		},
-		enabled: activeTab === 1 && stakedEscrowedBalance.gt(0) && !!parseFloat(amount),
-	});
-
-	const { writeAsync: kwentaApprove } = useContractWrite(kwentaApproveConfig);
-	const { writeAsync: stakeEscrowKwenta } = useContractWrite(stakedEscrowKwentaConfig);
-	const { writeAsync: unstakeEscrowKwenta } = useContractWrite(unstakedEscrowKwentaConfig);
+	const handleTabChange = useCallback((tabIndex: number) => {
+		setAmount('');
+		setActiveTab(tabIndex);
+	}, []);
 
-	const submitEscrow = useCallback(async () => {
-		if (kwentaTokenApproval) {
-			const approveTxn = await kwentaApprove?.();
-			monitorTransaction({
-				txHash: approveTxn?.hash ?? '',
-			});
+	const isDisabled = useMemo(() => {
```

Comment:
Would be good to also disable these actions buttons while a transaction is pending and show a loading status. Can use the pattern of storing transaction state in staking state.